src/bcrypt/__init__.py

```diff
@@ -78,6 +79,27 @@ def hashpw(password, salt):
     return _bcrypt.ffi.string(hashed)
 
 
+def checkpw(password, hashed_password):
+    if (isinstance(password, six.text_type) or
+            isinstance(hashed_password, six.text_type)):
+        raise TypeError("Unicode-objects must be encoded before checking")
+
+    if b"\x00" in password or b"\x00" in hashed_password:
+        raise ValueError(
+            "password and hashed_password may not contain NUL bytes"
+        )
+
+    # If the user supplies a $2y$ prefix we normalize to $2b$
+    hashed_password = _normalize_prefix(hashed_password)
+
+    try:
+        ret = hashpw(password, hashed_password)
+    except ValueError:
```

Comment:
Per your comment, you're totally right about hashpw only raising invalid salt if the salt itself has an issue so I agree we should probably just let this propagate.